assets/js/components/Dashboard/PaMessagesPage.tsx

```diff
@@ -243,16 +288,35 @@ const PaMessageTable: ComponentType<PaMessageTableProps> = ({
 
 interface PaMessageRowProps {
   paMessage: PaMessage;
+  stateFilter: StateFilter;
+  updatePaMessage: (paMessage: PaMessage) => void;
+  setErrorMessage: (message: string | null) => void;
 }
 
 const PaMessageRow: ComponentType<PaMessageRowProps> = ({
   paMessage,
+  stateFilter,
+  updatePaMessage,
+  setErrorMessage,
 }: PaMessageRowProps) => {
   const navigate = useNavigate();
   const start = new Date(paMessage.start_datetime);
   const end =
     paMessage.end_datetime === null ? null : new Date(paMessage.end_datetime);
 
+  const togglePaused = async (event: React.MouseEvent<HTMLDivElement>) => {
+    event.stopPropagation();
+    const result = await updateExistingPaMessage(paMessage.id, {
+      paused: !paMessage.paused,
+    } as UpdatePaMessageBody);
+
+    if (result.status === 200) {
```

Comment:
This is better, but it still doesn't catch network errors that cause the request to fail entirely. The way I usually like to handle this is to modify the API function (`updateExistingPaMessage` in this case) to include something like:
```js
// using ok allows any successful response, e.g. 201
if (!response.ok) {
  const error = new Error("request failed")
  // attach any available error information from the server
  throw error
}
```
Now the API function's promise will resolve on success and reject on any failure, so you can use it like:
```js
try {
  const payload = await updateExistingPaMessage(...)
  // do happy path stuff
}
catch (error) {
  // display error message
}
```

Reply:
Thinking of the other usage of this API function on the EditPaMessage page if we modified the function this way, how would we attach the list of errors to the returned `Error` object when the server returns that? It doesn't seem like `Error` has a property that lets us easily pass along the response body.

Reply:
There are two approaches, and it kind of depends on how sophisticated the error reporting needs to be. If we just want to show a little diagnostic message, it might be enough to just read/parse the body and stick it in as the `message` string on the `Error`, and then just display that. If we need something fancier, then we could attach arbitrary data to the `Error` object by just assigning a custom property. The latter probably isn't needed, but I just wanted to mention it.

Reply:
This seems good for now. If we like this pattern and want to apply it more generally in the future, we can formalize things by e.g. adding custom error subtypes with known data shapes, but that's probably premature at this point.